### PR TITLE
Added server autobot that crawls an index and build scenarios

### PR DESCRIPTION
**Pull request by adambbolduc**

### CODE REVIEW

File: autobot/autobot.go [hunk 1]
```diff
@@ -0,0 +1,96 @@
+package autobot
+
+import (
+	"github.com/erocheleau/uabot/explorerlib"
+	"github.com/erocheleau/uabot/scenariolib"
```
Comment (lbenie) [inline autobot/autobot.go]: github.com/**coveo**/uabot/scenariolib

File: autobot/autobot.go [hunk 2]
```diff
@@ -0,0 +1,96 @@
+package autobot
+
+import (
+	"github.com/erocheleau/uabot/explorerlib"
```
Comment (lbenie) [inline autobot/autobot.go]: github.com/**coveo**/uabot/explorerlib

File: explorerlib/configBuilder.go [hunk 3]
```diff
@@ -0,0 +1,90 @@
+package explorerlib
+
+import (
+	"encoding/json"
+	"github.com/erocheleau/uabot/scenariolib"
```
Comment (lbenie) [inline explorerlib/configBuilder.go]: change the path to 

github.com/**coveo**/uabot/scenariolib

File: explorerlib/scenarioBuilder.go [hunk 4]
```diff
@@ -0,0 +1,87 @@
+package explorerlib
+
+import (
+	"github.com/erocheleau/uabot/scenariolib"
```
Comment (lbenie) [inline explorerlib/scenarioBuilder.go]: same here

File: server.go [hunk 5]
```diff
@@ -0,0 +1,39 @@
+package main
+
+import (
+	"flag"
+	"fmt"
+	"github.com/erocheleau/uabot/server"
```
Comment (lbenie) [inline server.go]: same here

File: server.go [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+	"github.com/erocheleau/uabot/scenariolib"
```
Comment (lbenie) [inline server.go]: same here

File: server/JsonConfigDecoder.go [hunk 7]
```diff
@@ -0,0 +1,13 @@
+package server
+
+import (
+	"encoding/json"
+	"github.com/erocheleau/uabot/explorerlib"
```
Comment (lbenie) [inline server/JsonConfigDecoder.go]: same here

File: server/README.md [hunk 8]
```diff
@@ -0,0 +1,42 @@
+# Autobot 
+
+Application wrapper to the exisiting [uabot](https://github.com/erocheleau/uabot) to send "intelligent" usage analytics to simulate user interaction with an organization. Autobot grant an endpoint service to POST an autobot query.
```
Comment (lbenie) [inline server/README.md]: same here

File: server/botWorker.go [hunk 9]
```diff
@@ -0,0 +1,39 @@
+package server
+
+import (
+	"github.com/erocheleau/uabot/autobot"
```
Comment (lbenie) [inline server/botWorker.go]: same here

File: server/botWorker.go [hunk 10]
```diff
@@ -0,0 +1,39 @@
...
+	"github.com/erocheleau/uabot/explorerlib"
```
Comment (lbenie) [inline server/botWorker.go]: same here

File: server/botWorker.go [hunk 11]
```diff
@@ -0,0 +1,39 @@
...
+	"github.com/erocheleau/uabot/scenariolib"
```
Comment (lbenie) [inline server/botWorker.go]: same here

File: server/handlers.go [hunk 12]
```diff
@@ -0,0 +1,126 @@
+package server
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"github.com/erocheleau/uabot/explorerlib"
```
Comment (lbenie) [inline server/handlers.go]: same here

File: server/handlers.go [hunk 13]
```diff
@@ -0,0 +1,126 @@
...
+	"github.com/erocheleau/uabot/scenariolib"
```
Comment (lbenie) [inline server/handlers.go]: same here

File: Scripts/Dockerfile [hunk 14]
```diff
@@ -0,0 +1,11 @@
+FROM golang
+
+#get source code
+RUN git clone https://github.com/adambbolduc/uabot.git
```
Comment (lbenie) [inline Scripts/Dockerfile]: might want to change this ?

what do you think @erocheleau

Comment (adambbolduc) [inline Scripts/Dockerfile]: You are right !

**Comment by erocheleau** [comment]
Need to fix build before merge.

**Comment by erocheleau** [comment]
Also you desperately need to comment the new stuff if you hope that one day I will be able to go through it :)

I don't know what IDE you use to code but a go linter would ask you to comment ANYTHING that is exposed (any function, variable, struct, constant, etc. that starts with a capital letter)

This is to also be able to generate stuff like this which is done automagically with go:
https://godoc.org/github.com/coveo/uabot/scenariolib

**Pull request closed without merging**